SparkPost/2web2ui
# TR-2923 - Add back download CSV button on template list page

[TR-2923](https://sparkpost.atlassian.net/browse/TR-2923)

### What Changed
- Incorporates "Save As CSV" link to the `/templates` page
- Removes unhelpful snapshot test
- Re-organizes some Cypress tests and incorporates test for new "Save As CSV" link

### How To Test

- Go to `/templates` and verify there is a "Save As CSV" link present
- Click it and make sure it downloads data for the table

### To Do
